hibernate-core/src/main/java/org/hibernate/NaturalIdMultiLoadAccess.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.hibernate.graph.GraphSemantic;
+import org.hibernate.graph.RootGraph;
+import org.hibernate.internal.util.collections.CollectionHelper;
+
+/**
+ * Defines the ability to load multiple entities by simple natural-id simultaneously.
+ */
+public interface NaturalIdMultiLoadAccess<T> {
+	/**
+	 * Specify the {@link LockOptions} to use when retrieving the entity.
+	 *
+	 * @param lockOptions The lock options to use.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> with(LockOptions lockOptions);
+
+	/**
+	 * Specify the {@link CacheMode} to use when retrieving the entity.
+	 *
+	 * @param cacheMode The CacheMode to use.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> with(CacheMode cacheMode);
+
+	/**
+	 * Define a load graph to be used when retrieving the entity
+	 */
+	default NaturalIdMultiLoadAccess<T> with(RootGraph<T> graph) {
+		return with( graph, GraphSemantic.LOAD );
```

Comment:
Any reason to choose `LOAD` as the default entity graph type? Seems `FETCH` is the default type as per JPA spec.

Reply:
Not sure what you mean.  The JPA spec says nothing about loading by natural-id, so how can either be "per spec"?